### PR TITLE
Reorganize style-spec repository

### PR DESCRIPTION
**Pull request by birkskyum**
The situation around the Style Spec is not as organized as it should be. The spec itself is not versioned, the utilities are including files to maplibre gl js which shouldn't be there, and there are things they can't export because it would blow up gl js bundle size. This PR is the first step towards resolving the situation by isolating the specification .json files, and having everything else, which is implementation code (.ts / .c++ etc.), separate.

The three root folders of the repo are style-spec, docs, utils.

`style-spec` only has .json files, and a CHANGELOG with changes related to the .json files - which parameters are allowed etc. The idea is that these files with be a github release bundle (a .tar / .zip something like that), and that each implementation will pull that in, instead of installing the big style utility npm package.

`docs` contain the documentation for the style-spec, and it does require a bit of implementation code, which from now comes from the utils package, but likely could come directly from maplibre-gl-js instead.

`utils` this is the npm package with style spec utilities, such as validators. Currently this package is included in GL JS, but the intention is to go the other way around, so this package will pull the structures it needs from GL JS instead, so we can expose as many functions as we like from this without increasing the render bundle size. This can solve a longstanding issue #24

### CODE REVIEW

File: utils/package.json [hunk 1]
```diff
@@ -44,6 +45,7 @@
     "@mapbox/point-geometry": "^0.1.0",
     "@mapbox/unitbezier": "^0.0.1",
     "@types/mapbox__point-geometry": "^0.1.2",
+    "copyfiles": "^2.4.1",
```
Comment (HarelM) [inline utils/package.json:48]: Dev dependency?

File: utils/src/style-spec.ts [hunk 2]
```diff
@@ -173,7 +171,7 @@ export {
 
     ColorType,
     interpolates,
-    v8,
+    latest as v8,
```
Comment (HarelM) [inline utils/src/style-spec.ts:174]: I think dts-bundle-generator has problems with exporting `something as somthingelse`...
I would reccomend simply importing it using `import v8 from './style-spec/latest'`.

File: utils/src/style-spec/latest.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+
+import spec from './specification.json';
```
Comment (HarelM) [inline utils/src/style-spec/latest.ts:2]: Should we use asset type json here or something?

**Comment by birkskyum** [comment]
In regards to this 

> There should be a folder related to running the tests in the maplibre-style-spec-gl/util folder with the relevant logic (may it be jest, puppeteer or what ever we choose in the future).

I think these package should only have the tests that aren't already covered in maplibre-gl, so that would be validation/style-spec conversion tests etc., but not i.e. expression tests

**Comment by HarelM** [comment]
@birkskyum is this PR still relevant? Do we want to push it through or close it?

**Comment by HarelM** [comment]
Closing for now as it looks like it will need a rework anyway.
Feel free to open a new one if this is still relevant.

**Pull request closed without merging**